contracts/test/unit/EigenDACertVerifierV3Unit.t.sol

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.12;
+
+import "../MockEigenDADeployer.sol";
+import {EigenDACertVerificationV2Lib as CertV2Lib} from "src/periphery/cert/v2/EigenDACertVerificationV2Lib.sol";
+import {EigenDATypesV2} from "src/core/libraries/v2/EigenDATypesV2.sol";
+import {EigenDATypesV1} from "src/core/libraries/v1/EigenDATypesV1.sol";
+import {EigenDACertTypes as CertTypes} from "src/periphery/cert/EigenDACertTypes.sol";
+import {EigenDACertVerifierV3} from "src/periphery/cert/v3/EigenDACertVerifierV3.sol";
+
+import {console2} from "forge-std/console2.sol";
+
+contract EigenDACertVerifierV2Unit is MockEigenDADeployer {
+    using stdStorage for StdStorage;
+    using BN254 for BN254.G1Point;
+
+    address relay0 = address(uint160(uint256(keccak256(abi.encodePacked("relay0")))));
+    address relay1 = address(uint160(uint256(keccak256(abi.encodePacked("relay1")))));
+
+    EigenDACertVerifierV3 certVerifierV3;
+
+    function setUp() public virtual {
+        quorumNumbersRequired = hex"00";
+        _deployDA();
+        certVerifierV3 = new EigenDACertVerifierV3(
+            eigenDAThresholdRegistry,
+            IEigenDASignatureVerifier(address(eigenDAServiceManager)),
+            defaultSecurityThresholds,
+            quorumNumbersRequired
+        );
+    }
+
+    function test_verifyDACertV3(uint256 pseudoRandomNumber) public {
+        (
+            EigenDATypesV2.SignedBatch memory signedBatch,
+            EigenDATypesV2.BlobInclusionInfo memory blobInclusionInfo,
+            BLSSignatureChecker.NonSignerStakesAndSignature memory nssas
+        ) = _getSignedBatchAndBlobVerificationProof(pseudoRandomNumber, 0);
+
+        EigenDATypesV1.NonSignerStakesAndSignature memory nonSignerStakesAndSignature;
+        nonSignerStakesAndSignature.nonSignerQuorumBitmapIndices = nssas.nonSignerQuorumBitmapIndices;
+        nonSignerStakesAndSignature.nonSignerPubkeys = nssas.nonSignerPubkeys;
+        nonSignerStakesAndSignature.quorumApks = nssas.quorumApks;
+        nonSignerStakesAndSignature.apkG2 = nssas.apkG2;
+        nonSignerStakesAndSignature.sigma = nssas.sigma;
+        nonSignerStakesAndSignature.quorumApkIndices = nssas.quorumApkIndices;
+        nonSignerStakesAndSignature.totalStakeIndices = nssas.totalStakeIndices;
+        nonSignerStakesAndSignature.nonSignerStakeIndices = nssas.nonSignerStakeIndices;
+
+        (DATypesV1.NonSignerStakesAndSignature memory _nonSignerStakesAndSignature, bytes memory signedQuorumNumbers) =
+            CertV2Lib.getNonSignerStakesAndSignature(operatorStateRetriever, registryCoordinator, signedBatch);
+
+        bytes memory certBytes = abi.encode(
```

Comment:
does abi encode packed work?

Reply:
It shouldn't work since encodePacked would not have the same padding that encode does. What is the context for this q? Are we trying to reduce calldata size? If we want to reduce calldata size, we would likely need to write a special library to handle decoding of a cert since there's nothing in solidity that would support a mixed decoding scheme. Maybe this can be future work?